projects/uart/README.md

```diff
@@ -0,0 +1,151 @@
+
+<img src="img/data_frame.jpg" alt="Data frame" title="Data frame">
+
+- a 16 bits request code
+- Data
+- a 16 bits CRC
+
+There are two possible transfer mechanisms. For each, there can be only one request at a time before receiving an answer.
+- Standard request: the transfer is initiated by the host and wait for an answer from the host adapter
+
+<img src="img/standard_request.jpg" alt="Data frame" title="Data frame">
+
+- Notification: the transfer is initiated by the host adapter and wait for an answer from the host
+
+<img src="img/notification.jpg" alt="Data frame" title="Data frame">
+
+<br/>
+
+**Request code**:
+
+Digital functions
+
+| Range start | Range end | Function              |
+| ----------- | --------- | --------------------- |
+| `0x0000`    | `0x00FF`  | Generic requests      |
+| `0x0100`    | `0x01FF`  | GPIO request codes    |
+| `0x0200`    | `0x02FF`  | PWM request codes     |
+| `0x0300`    | `0x03FF`  | Encoder request codes |
+| `0x0400`    | `0x04FF`  | Timer request codes   |
+
+ Protocols
+
+| Range start | Range end | Function   |
+| ----------- | --------- | ---------- |
+| `0x3800`    | `0x38FF`  | UART       |
+| `0x3900`    | `0x39FF`  | SPI master |
+| `0x3A00`    | `0x3AFF`  | SPI slave  |
+| `0x3B00`    | `0x3BFF`  | I2C master |
+| `0x3C00`    | `0x3CFF`  | I2C slave  |
+| `0x3D00`    | `0x3DFF`  | Modbus RTU |
+| `0x3E00`    | `0x3EFF`  | *TBD*      |
+| `0x3F00`    | `0x3FFF`  | *TBD*      | 
+| `0x4000`    | `0x40FF`  | CAN 2.0    |
+| `0x4100`    | `0x41FF`  | CAN FD     |
+| `0x4200`    | `0x42FF`  | LIN        |
+
+Digital functions answers
+
+| Range start | Range end | Function              |
+| ----------- | --------- | --------------------- |
+| `0xFFFF`    | `0xFF00`  | Shared answers        |
+| `0xFEFF`    | `0xFE00`  | Generic answers       |
+| `0xFDFF`    | `0xFD00`  | GPIO answers codes    |
+| `0xFCFF`    | `0xFC00`  | PWM answers codes     |
+| `0xFBFF`    | `0xFB00`  | Encoder answers codes |
+| `0xFAFF`    | `0xFA00`  | Timer answers codes   |
+
+ Protocols answers
+
+| Range start | Range end | Function   |
+| ----------- | --------- | ---------- |
+| `0xB4FF`    | `0xB400`  | UART       |
+| `0xB3FF`    | `0xB300`  | SPI master |
+| `0xB2FF`    | `0xB200`  | SPI slave  |
+| `0xB1FF`    | `0xB100`  | I2C master |
+| `0xB0FF`    | `0xB000`  | I2C slave  |
+| `0xAFFF`    | `0xAF00`  | Modbus RTU |
+| `0xAEFF`    | `0xAE00`  | *TBD*      |
+| `0xADFF`    | `0xAD00`  | *TBD*      | 
+| `0xACFF`    | `0xAC00`  | CAN 2.0    |
+| `0xABFF`    | `0xAB00`  | CAN FD     |
+| `0xAAFF`    | `0xAA00`  | LIN        |
+
+Generic requests
+
+| Code        | Function  |
+| ----------- | --------- |
+| `0x0000`    | Ping      |
+| `0x0001`    | ItfType   |
+| `0x0002`    | Version   |
+| `0x0003`    | IdGet     |
+
+### SECTION 2: Features
+
+#### [REQ_2000] Send data
+
+The product **must** ....
+
+#### [REQ_2010] Baud rate
+
+The product **must** enable the baud rate configuration
```

Comment:
enable the baudrate ? 
Plutot
must provide a way to change and read the baudrate ?
Ensuite tu peux ajouter ici le 'code' qui sera a écrire le baudrate et le code pour lire (ces spécifications vont être très orienté conception)